File: TUF/String/PrintAllDuplicates.ts
function printAllDuplicates(str: string): void {
    const map = new Map<string, number>();
    for (const char of str) {
        map.set(char, (map.get(char) || 0) + 1);
    }

    for (const [key, value] of map) {
        if (value > 1) {
            console.log(`${key} --> ${value}`);
        }
    }
}

let str: string = "sinstriiintng";
printAllDuplicates(str);

export { };

File: TUF/String/PrintAllDuplicates.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class PrintAllDuplicates {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the string: ");
        String str = input.next();
        input.close();
        printAllDuplicates(str);
    }

    private static void printAllDuplicates(String str) {
        Map<Character, Integer> map = new HashMap<>();
        for (int i = 0; i < str.length(); i++) {
            char currChar = str.charAt(i);
            map.put(currChar, map.getOrDefault(currChar, 0) + 1);
        }

        for (Map.Entry<Character, Integer> entry : map.entrySet()) {
            if (entry.getValue() <= 1) {
                continue;
            }
            System.out.println(entry.getKey() + " --> " + entry.getValue());
        }
    }
}
